### DISCUSSION TITLE
Why do the stdev and var aggregate functions not ignore null values like sum, min, max, and avg ?

### DISCUSSION THREAD

**Discussion by BlancoMY** [lf-edge/ekuiper#2589] [Q&A] [answered]
Hello,

I've noticed that while using eKuiper, the stdev and var aggregate functions seem to consider null values in their calculations, unlike other aggregate functions like sum, min, max, and avg which ignore null values. Can anyone shed light on the reason behind this behavior and whether there are any workarounds or alternative approaches to handle null values with stdev and var functions in eKuiper

Thank you

**Comment by ngjaying** [comment, marked as answer]
Looks like a defect
